python/nammu/controller/AtfAreaController.py

```diff
@@ -182,3 +183,16 @@ def restore_highlight(self):
         length = len(self.getAtfAreaText())
         self.syntax_highlighter._highlight_match(0, length, Color.white)
         self.syntax_highlighter.syntax_highlight()
+
+    def getPositionFromLine(self, text, line_num):
+        '''
+        Given a block of text and a line number, return a tuple containing
+        the caret position at the start and end of the given line.
+        '''
+        compiled = re.compile(r"\n")
+        textiter = compiled.finditer(text)
+        if line_num != 1:
+            pos = [m.start() for m in textiter][line_num - 2:line_num]
+        else:
+            pos = [0, textiter.next().start()]
+        return pos
```

Comment:
This has been done to reflect behaviour in `SyntaxHighlighter.py` in an attempt to begin refactoring some of that code. This turned out to be a bigger challenge than initially thought, so a new issue (#239) has been opened to do this.

The code works at present and these additional changes will be made in a new branch following the merge of this pr.